### PR TITLE
Add CircuitPython demo

### PR DESCRIPTION
**Pull request by tekktrik**
Adds a demo for CircuitPython, instructions on creating a FAT12 filesystem, and updated instructions in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,6 +74,39 @@ Other ways of creating LittleFS images can be found [here](https://github.com/wo
...
+### CircuitPython code
+
+To run the CircuitPython demo, you can follow the directions above for MicroPython, except download [adafruit-circuitpython-raspberry_pi_pico-en_US-8.0.2.uf2](https://adafruit-circuit-python.s3.amazonaws.com/bin/raspberry_pi_pico/en_US/adafruit-circuitpython-raspberry_pi_pico-en_US-8.0.2.uf2) instead of the MicroPython UF2 file. Place it in the rp2040js root directory, then run:
+
+```
+npm install
+npm run start:circuitpython
+```
+
+and start the CircuitPython REPL! The rest of the experience is the same as the MicroPython demp (Ctrl+X to exit, using the `--image` and
```
Comment (urish) [inline README.md:86]: typo: demp -> demo

File: demo/micropython-run.ts [hunk 2]
```diff
@@ -12,20 +12,30 @@ const args = minimist(process.argv.slice(2), {
     'image', // UF2 image to load; defaults to "rp2-pico-20210902-v1.17.uf2"
     'expect-text', // Text to expect on the serial console, process will exit with code 0 if found
   ],
-  boolean: 'gdb', // start GDB server on 3333
+  boolean: [
+    'gdb', // start GDB server on 3333
+    'circpy', // use CircuitPython instead of MicroPython
```
Comment (urish) [inline demo/micropython-run.ts:17]: why not use the full name, `circuitpython`?

**Review by urish — CHANGES_REQUESTED** [review]

File: demo/micropython-run.ts (hunk 2 above)
Comment (tekktrik) [reply demo/micropython-run.ts:17]: No reason, I tend to default to shorter flag names but I can change it - seems more sensible in full!

Comment (urish) [reply demo/micropython-run.ts:17]: Yes please

**Pull request merged**